SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/EventTracingTests/ProcessCrossServerRPCEventTracingTest.cpp

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "ProcessCrossServerRPCEventTracingTest.h"
+
+AProcessCrossServerRPCEventTracingTest::AProcessCrossServerRPCEventTracingTest()
+{
+	Author = "Danny Birch";
+	Description = TEXT("Test checking the process RPC trace events have appropriate causes for cross-server RPCs");
+
+	FilterEventNames = { ReceiveCrossServerRPCName, ApplyCrossServerRPCName };
+	WorkerDefinition = FWorkerDefinition::Server(1);
+}
+
+void AProcessCrossServerRPCEventTracingTest::FinishEventTraceTest()
+{
+	int EventsTested = 0;
+	int EventsFailed = 0;
+	for (const auto& Pair : TraceEvents)
+	{
+		const FString& SpanIdString = Pair.Key;
+		const FName& EventName = Pair.Value;
+
+		if (EventName == ReceiveCrossServerRPCName)
+		{
+			continue;
+		}
+
+		EventsTested++;
+
+		if (EventName == ApplyCrossServerRPCName)
+		{
+			if (!CheckEventTraceCause(SpanIdString, { ReceiveCrossServerRPCName }, true))
+			{
+				EventsFailed++;
+			}
+		}
+		// 		else // EventName == ReceiveRPCEventName
+		// 		{
+		// 			if (!CheckEventTraceCause(SpanIdString, { ReceiveOpEventName }, true))
+		// 			{
+		// 				EventsFailed++;
+		// 			}
+		// 		}
+	}
+
+	bool bSuccess = EventsTested > 0 && EventsFailed == 0;
```

Comment:
How come not a specific number for `EventsTested`?

Reply:
It's a timed scenario. If the time changes the number of events changes. Would be brittle if we had a set number.